packages/vite/src/node/plugins/worker.ts

```diff
@@ -276,10 +275,23 @@ export function webWorkerPlugin(config: ResolvedConfig): Plugin {
         }`,
         map: { mappings: '' } // Empty sourcemap to supress Rolup warning
       }
-    },
+    }
+  }
+}
+
+// just run in build mode
+export function webWorkerImportMetaUrlPlugin(config: ResolvedConfig): Plugin {
+  const isWorker = config.isWorker
 
-    renderChunk(code) {
-      if (isWorker && code.includes('import.meta.url')) {
+  return {
+    name: 'vite:workerImportMetaUrl',
+    transform(code) {
+      if (isWorker && config.worker.format !== 'es') {
+        // if build with iife it will polyfill with rollup will be used document in the worker
+        // else if build with es it will replace with `define plugin` will got a unexpected data.
+        // so replace import.meta.url break the default handle.
+        // And it must be done at this (after `(?new Worker)new URL('xxx', import.meta.url)` match),
+        // Otherwise, the following regex will be incorrectly matched.
         return code.replace('import.meta.url', 'self.location.href')
```

Comment:
Does this replaces every instance if there are more than one in the file? I saw you reverted `replaceAll`, maybe we should use a `replace` with a global regex?

Reply:
well, may be no. I saw the need for es2021 support `replaceAll` and thought they were the same. 🤦

```text
> "a  a a aa a a".replace("a", "#")
'#  a a aa a a'
> "import.meta.url import.meta.urla 1import.meta.url".replace(/\bimport.meta.url\b/g, 'self.location.href')
'self.location.href import.meta.urla 1import.meta.url'
```